kernel/tests/write.rs

```diff
@@ -139,136 +149,174 @@ fn new_commit_info() -> DeltaResult<Box<ArrowEngineData>> {
     Ok(Box::new(ArrowEngineData::new(commit_info_batch)))
 }
 
```

Comment:
It looks like there are a lot of changes in this file, but mainly it's just:
1. Allowing`setup` to create a 3/7 or 1/1 table
2. adding a new function to generate both types of tables and associated store/engine/location
3. making each test iterate over those tables instead of just one